realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -2281,4 +2309,127 @@ public void distinctAsync_notIndexedLinkedFields() {
         } catch (IllegalArgumentException ignored) {
         }
     }
+
+    @Test
+    public void distinctMultiArgs() {
+        final long numberOfBlocks = 25;
+        final long numberOfObjects = 10; // must be greater than 1
+        populateForDistinct(realm, numberOfBlocks, numberOfObjects, false);
+
+        //Simon Ask Ulsnes "I would expect it to be 25, because I would expect it to look for unique ​_combinations_​ or tuples."
+        RealmResults<AnnotationIndexTypes> distinctMulti = realm.where(AnnotationIndexTypes.class).distinct(AnnotationIndexTypes.INDEX_FIELDS);
```

Comment:
These doesn't test RealmQuery distinct but Realm distinct? Same for all the tests? That they use RealmQuery is an implementation detail that should be encoded into the tests